### PR TITLE
Img to next image

### LINKED ISSUE #4382: Add next/image component as MDX component and convert img tags to Image component
There are a lot of images inside markdown that are either `img` tags or markdown images. Need to use next/image component instead of img tags and markdown images.

### PR DESCRIPTION
**Pull request by prakhart4**
Closes #4382 

**PR contains:**
* Replacing `` tags with the Next image component.
* Adding `height` and `width` in image props, if not available.

### CODE REVIEW

File: src/components/MDX/MDXComponents.tsx [hunk 1]
```diff
@@ -368,8 +370,28 @@ function YouTubeIframe(props: any) {
...
+  return (
+    <div
+      style={{
+        position: 'relative',
+        width: !width ? '' : `${width}px`,
+        maxHeight: !height ? '' : `${height}px`,
+        maxWidth: 'calc(min(700px,100%))',
+        ...(!style ? {} : style),
+      }}
+      className={styles.imageContainer}>
```
Comment (harish-sethuraman) [inline src/components/MDX/MDXComponents.tsx:384]: can we rather use tailwind classes? + move the `position: 'relative'` also to classname

File: src/components/MDX/MDXComponents.tsx [hunk 2]
```diff
@@ -368,8 +370,28 @@ function YouTubeIframe(props: any) {
...
+      <NextImage
+        {...props}
+        src={
+          props?.src?.startsWith('/') ? props.src : props.src.replace('..', '')
+        }
+        layout={'fill'}
+        className={styles.image}
```
Comment (harish-sethuraman) [inline src/components/MDX/MDXComponents.tsx:391]: can we use tailwind classname here as well?

File: src/components/MDX/MDXComponents.tsx (hunk 1 above)
Comment (prakhart4) [reply src/components/MDX/MDXComponents.tsx]: Yes, using tailwind classes now. I was not able to target the nextImage wrapper using tailwind v3.0, now added a variable NextImageWrapper in the tailwind config.

File: src/components/MDX/MDXComponents.tsx (hunk 2 above)
Comment (prakhart4) [reply src/components/MDX/MDXComponents.tsx]: Sure

File: src/components/MDX/MDXComponents.tsx [hunk 3]
```diff
@@ -368,8 +369,28 @@ function YouTubeIframe(props: any) {
...
-function Image(props: any) {
-  return <img className="max-w-[calc(min(700px,100%))]" {...props} />;
+function Image(allProps: any) {
+  const {height, width, style, ...props} = allProps;
+  return (
+    <div
+      style={{
+        ...(!style ? {} : style),
+        ...(!width ? {} : {width: `${width}px`}),
+        ...(!height ? {} : {maxHeight: `${height}px`}),
```
Comment (harish-sethuraman) [inline src/components/MDX/MDXComponents.tsx:379]: actually you could remove the `!` and swap the conditions too :p

File: src/components/MDX/MDXComponents.tsx [hunk 4]
```diff
@@ -368,8 +369,28 @@ function YouTubeIframe(props: any) {
...
+      }}
+      className={
+        'relative max-w-[calc(min(700px,100%))] NextImageWrapper:!static'
+      }>
```
Comment (harish-sethuraman) [inline src/components/MDX/MDXComponents.tsx:383]:
```suggestion
      className="relative max-w-[calc(min(700px,100%))] NextImageWrapper:!static">
```

File: tailwind.config.js [hunk 5]
```diff
@@ -147,5 +147,8 @@ module.exports = {
       },
     },
   },
-  plugins: [],
+  plugins: [
+    function ({ addVariant }) {
+        addVariant('NextImageWrapper', '& > span');
```
Comment (harish-sethuraman) [inline tailwind.config.js:152]: add kebab case?

**Comment by harish-sethuraman** [comment]
cc: @gaearon lets track and merge this? its a long pending one thanks :)

File: src/components/MDX/MDXComponents.tsx (hunk 4 above)
Comment (prakhart4) [reply src/components/MDX/MDXComponents.tsx]: Changed all the {"..."} to "..."

File: src/components/MDX/MDXComponents.tsx (hunk 3 above)
Comment (prakhart4) [reply src/components/MDX/MDXComponents.tsx]: Done

**Comment by prakhart4** [comment]
Should I create a fresh PR?

**Comment by prakhart4** [comment]
@gaearon Please review the PR

**Comment by prakhart4** [comment]
Created fresh PR => Replaced Img with next Image #6263

**Pull request closed without merging**